### PR TITLE
Mobile Pdf links

### PR DESCRIPTION
**Pull request by gestchild**
## What does this change?

Relates to [#11862](https://github.com/wellcomecollection/wellcomecollection.org/issues/11862)

Puts in the functionality ahead of design being completed [12024](https://github.com/wellcomecollection/wellcomecollection.org/issues/12024).

Displays links for PDFs in the viewer for mobile devices instead of the iframe.

- adds an isMobileOrTablet property to the AppContext. Uses user agent sniffing - don't love it but think it is necessary. Would like opinions.

## How to test

- Enable the extendedViewer toggle
- Visit [an item page with PDFs](https://www-dev.wellcomecollection.org/works/zu2q4k2w/items)
- On desktop you should still see the PDF
- Open dev tools and change the user agent to a mobile device
Screenshot 2025-06-17 at 17 43 41
- The page should now show a link for the ODF instead of the iframe

## How can we measure success?

Mobile users can access the PDFs

## Have we considered potential risks?

behind a toggle so think this is ok to go ahead of designs being complete

### CODE REVIEW

**Comment by rcantin-w** [comment]
> adds an isMobileOrTablet property to the AppContext. Uses user agent sniffing - don't love it but think it is necessary. Would like opinions.

I had a look this morning because I also don't know and came across this [MDN doc: Browser detection using the user agent string (UA sniffing)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/Browser_detection_using_the_user_agent)

One of the examples brought me to [`navigator.isPDFinline`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/pdfViewerEnabled) and I wonder if that check would cover it? It seems promising but I only tested on Chrome/mocking a mobile

**Comment by gestchild** [comment]
Thanks @rcantin-w I'm much more comfortable with that approach and seems to be working as expected

**Comment by gestchild** [comment]
hmm, just managed to get browser stack working and pdfViewerEnabled on iOS Safari is true (it wasn't when I was changing the user agent in the network tab), so we can't use it as the check in this case

**Comment by gestchild** [comment]
@rcantin-w I had to amend the string again to account for Tablet devices. This is what Claude suggested...

File: common/contexts/AppContext/index.tsx [hunk 1]
```diff
@@ -47,6 +49,20 @@ export function useAppContext(): AppContextProps {
...
+// Ideally we'd use feature detection to determine if the browser supports
+// displaying pdfs inline (navigator.pdfViewerEnabled).
+// However, iOS passes this test but does not provide a good user experience
+// We therefore use a user-agent based detection for mobile and tablet devices.
+function checkForMobileOrTabletDevice(): boolean {
+  return (
+    /Mobi|Android|iPad|iPhone|iPod|tablet|Tablet|Silk|Kindle|BlackBerry|PlayBook|KFAPWI|(Android.+Mobile)/i.test(
+      navigator.userAgent
+    ) ||
+    // Alternative approach to detect iPad with iOS 13+ that disguises as desktop
```
Comment (rcantin-w) [inline common/contexts/AppContext/index.tsx:61]: cheeky

**Review by rcantin-w — APPROVED** [review]
OK I think let's try it, we can test better when it's deployed.

None of us want to use that too often anyway in the future, and this feels edge case-y enough to get it out and trial it!
